cg/apps/demultiplex/sample_sheet/read_sample_sheet.py

```diff
@@ -2,14 +2,14 @@
 
 from pydantic import TypeAdapter
 
-from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample
+from cg.apps.demultiplex.sample_sheet.sample_models import IlluminaSampleIndexSettings
 from cg.constants.demultiplexing import SampleSheetBcl2FastqSections, SampleSheetBCLConvertSections
 from cg.exc import SampleSheetContentError, SampleSheetFormatError
 
 LOG = logging.getLogger(__name__)
 
 
-def validate_samples_are_unique(samples: list[FlowCellSample]) -> None:
+def validate_samples_are_unique(samples: list[IlluminaSampleIndexSettings]) -> None:
```

Comment:
Looks like a list with a list of IlluminaIndexSetting objects to me due to the plural.

Reply:
Other option would be `IlluminaSampleIndex`